src/stratis_cli/_parser/_parser.py

```diff
@@ -23,44 +23,81 @@
 
 from .._version import __version__
 
-from ._logical import build_logical_parser
-from ._physical import build_physical_parser
-from ._pool import build_pool_parser
+from ._logical import LOGICAL_SUBCMDS
+from ._physical import PHYSICAL_SUBCMDS
+from ._pool import POOL_SUBCMDS
 
-
-def build_daemon_parser(parser):
+def add_args(parser, args=None):
     """
-    Generates the parser appropriate for obtaining information about stratisd.
-
-    :param ArgumentParser parser: a parser
-    :returns: a completed parser for obtaining information about stratisd
-    :rtype: ArgumentParser
+    Call subcommand.add_argument() based on args list.
     """
-    subparsers = \
-       parser.add_subparsers(dest='subparser_name', title='subcommands')
-
-    redundancy_parser = subparsers.add_parser(
-       'redundancy',
-       description="redundancy designations understood by stratisd daemon"
-    )
-    redundancy_parser.set_defaults(func=StratisActions.list_stratisd_redundancy)
-
-    version_parser = subparsers.add_parser(
-       'version',
-       description="version of stratisd daemon"
-    )
-    version_parser.set_defaults(func=StratisActions.list_stratisd_version)
-
-    return parser
-
+    if args is not None:
+        for name, arg in args:
+            parser.add_argument(name, **arg)
 
-_SUBPARSER_TABLE = {
-   'blockdev' : build_physical_parser,
-   'filesystem' : build_logical_parser,
-   'pool' : build_pool_parser,
-   'daemon' : build_daemon_parser
-}
 
+def add_subcommand(subparser, cmd):
+    """
+    Add subcommand to a parser based on a subcommand dict.
+    """
+    name, info = cmd
+    parser = subparser.add_parser(name, help=info['help'])
+
+    subcmds = info.get('subcmds')
+    if subcmds:
```

Comment:
Please do explicit comparison with None, as ```if subcmds is not None:``` and same on line 54. It should never make any difference here, but I've seen overloading in Python leading to an enormous number of bugs, so I'm only in favor of it when it is needeed.